### PR TITLE
Explicitly return error incase of not able to set provider id in correct format

### PR DESCRIPTION
**Pull request by Karthik-K-N**
**What this PR does / why we need it**:

This PR modifies the logic around setting provider id.
There were many instances where provider id was not set properly and the cluster creation was not successfull. Main reason for that is service instance id being set empty.

Made changes to fetch service instance id from name if id not found in spec or status. This case will occur mainly when cluster is reconciled by external controller.
Made changes to return error instead of empty string in case of not able to find service instance id.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by Karthik-K-N** [comment]
/hold
Ready for review but hold for some testing.

File: cloud/scope/powervs_machine.go [hunk 1]
```diff
@@ -934,29 +934,45 @@ func (m *PowerVSMachineScope) GetZone() string {
...
+	if m.IBMPowerVSCluster.Spec.ServiceInstance != nil && m.IBMPowerVSCluster.Spec.ServiceInstance.Name == nil {
+		return "", fmt.Errorf("failed to find service instance id as both name and id are not set")
+	}
+	serviceInstance, err := m.ResourceClient.GetServiceInstance("", *m.IBMPowerVSCluster.Spec.ServiceInstance.Name, ptr.To(m.GetZone()))
+	if err != nil {
+		m.Error(err, "failed to get Power VS service instance id", "serviceInstanceName", *m.IBMPowerVSCluster.Spec.ServiceInstance.Name)
+		return "", err
+	}
+	// It's safe to directly dereference GUID as its already done in NewPowerVSMachineScope
+	return *serviceInstance.GUID, nil
```
Comment (mkumatag) [inline cloud/scope/powervs_machine.go:957]: but we have removed this code intentionally before right?

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: cloud/scope/powervs_machine.go (hunk 1 above)
Comment (mkumatag) [reply cloud/scope/powervs_machine.go:957]: @Amulyam24 ^^

Comment (Amulyam24) [reply cloud/scope/powervs_machine.go:957]: @mkumatag, this a different function called using PowerVSMachineScope
The function we modified earlier is https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/main/cloud/scope/powervs_cluster.go#L388 called using PowerVSClusterScope

**Comment by Karthik-K-N** [comment]
/unhold
Verified creating a CAPI cluster and machine provider id was set as expected.

**Review by mkumatag — APPROVED** [review]

**Pull request merged**